android/engine/src/test/resources/plans/child-immunization-schedule/structure-map-register.txt

```diff
@@ -58,7 +88,18 @@ group ExtractTask(
         subject then {
                  subject -> evaluate(careplan, activity.where(detail.kind = 'Task')) as activity,
                             activity.outcomeReference = reference(task) "r_cp_task_ref";
-                 subject -> careplan.contained = task  "r_add_task";
+                 subject -> careplan.contained = task "r_add_task";
         } "r_cp_task";
+
+        // partof
+         subject-> evaluate(definition,$this.product.text + ' ' +$this.dosage.skip(version).first().select((sequence-1).toString())) as preReqDosage then{// OPV 1
+                   subject -> task.partOf = create('Reference') as ref then {
+                              subject -> ref.reference = evaluate(careplan,$this.contained.where(input.last().value =preReqDosage).first().id) "r_task_ref_id";
```

Comment:
@ahsanbhatti49 I think the `partOf` attribute should be added for the dependent tasks of the same dosage.

Reply:
This piece of code just add the initially created task as the `partOf`